src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -380,6 +382,9 @@ protected void ExecuteIndexing()
 
                             DoIndexingWork(stats, cts.Token);
 
+                            _indexingBatchCompleted.SetByAsyncCompletion();
+                            _indexingBatchCompleted.Reset();
+
```

Comment:
Auto reset event wouldn't work here because (copied from this article) " if four threads wait on an auto-reset event, and someone else sets the event, just one of the four waiters will complete, and the other three waiters will remain waiting until more signals arrive"